pkg/discovery/service.go

```diff
@@ -24,39 +24,49 @@ var (
 	Error = errs.Class("discovery error")
 )
 
-// Config loads on the configuration values from run flags
+// Config loads on the configuration values for the cache
 type Config struct {
 	RefreshInterval time.Duration `help:"the interval at which the cache refreshes itself in seconds" default:"1s"`
+	RefreshLimit    int           `help:"the amount of nodes refreshed at each interval" default:"100"`
+}
+
+// Refresh tracks the offset of the current refresh cycle
+type Refresh struct {
+	offset int64
 }
 
 // Discovery struct loads on cache, kad, and statdb
 type Discovery struct {
-	log    *zap.Logger
-	cache  *overlay.Cache
-	kad    *kademlia.Kademlia
-	statdb statdb.DB
-
-	refreshInterval time.Duration
+	log     *zap.Logger
+	cache   *overlay.Cache
+	kad     *kademlia.Kademlia
+	statdb  statdb.DB
+	config  Config
+	refresh Refresh
```

Comment:
i know this probably feels like nitpicking but why is this another struct? why not just do `refreshOffset int64` in line here?